common/gnome-shell/3.18/sass/_common.scss

```diff
@@ -964,12 +964,13 @@ StScrollBar {
     }
 
     .system-status-icon { icon-size: 16px; padding: 0 4px; }
-    .unlock-screen &,
-    .login-screen &,
-    .lock-screen & {
-      color: lighten($fg_color, 10%);
-      &:focus, &:hover, &:active { color: lighten($fg_color, 10%); }
-    }
+  }
+
+  &.unlock-screen .panel-button,
+  &.login-screen .panel-button,
+  &.lock-screen .panel-button {
+    color: lighten($fg_color, 10%);
+    &:focus, &:hover, &:active, &:checked { color: lighten($fg_color, 10%); }
   }
```

Comment:
Coding style wise, I think it would be much cleaner to move this into the already existing ruleset starting at line `903`.

However, I think for the Arc theme, it would probably make sense to not re-style the text color for login and lock screens. I believe the login/lock screen top panel background for Arc is always the dark panel color from `panel.svg` regardless of the variant? If that's the case, using the lightened `$fg_color` is wrong, since it is different for light and dark variants.

Using `$_panel_fg_color`, and $selected_fg_color for `:hover`, `:active` etc, should be correct, and this is already done at lines `937` and `957`. Therefore I think the `color` definitions could be simply removed from this ruleset altogether.

Reply:
Transparent button border for login and unlock screen (and probably for the lock screen as well) makes sense, but as I said earlier, it could be moved inside the existing ruleset above.

Reply:
> Coding style wise, I think it would be much cleaner to move this into the already existing ruleset starting at line `903`.

Done, although I separated the `.unlock-screen` and `.login-screen` rulesets from the `.lock-screen` ruleset

> The panel is transparent for `unlock-screen` and `login-screen`. I couldn't figure out the login screen background color from the code, but based on your screenshot, it should be `$bg_color`. In that case the `color` should be set to `$fg_color`:

Done

> The panel background for `.lock-screen` [appears to be](https://github.com/jnsh/arc-theme/blob/master/common/gnome-shell/3.32/sass/_common.scss#L2485) transparentized `$osd_bg_color`. Therefore `$osd_fg_color` would make sense for the lock-screen:

Done, and moved the code for the panel background in `.lock-screen` to the main `#panel` block of code

> The `&:focus, &:hover, &:active, &:checked` color should always be `$selected_fg_color` since `$selected_bg_color` is used as a background.

Done

> Transparent button border for login and unlock screen (and probably for the lock screen as well) makes sense, but as I said earlier, it could be moved inside the existing ruleset above.

Done

Additionally, I have modified the background of the unlock/login/lock screens (defined in `#lockDialogGroup`) as previously background was defined with a hardcoded color and reference to a non-existent resource

> could you post full screenshots for the login and lock screens with these color values? Additionally, it would be good to test that the panel-button `color` transitions smoothly between login and lock screens, and make sure that the light variant works also as expected.

I will post screenshots in a little while

Reply:
Screenshots

- **Lock screen (light/dark variants are the same):**

  ![lock](https://user-images.githubusercontent.com/18715287/61996370-ef1f5580-b093-11e9-91c7-0d0850d34f17.png)

- **Login/unlock screens (light variant):**

  ![login-light](https://user-images.githubusercontent.com/18715287/61996392-21c94e00-b094-11e9-875b-ed471f60d26f.png)
  ![login-light-hover](https://user-images.githubusercontent.com/18715287/61996393-255cd500-b094-11e9-9c69-b568f9aa4229.png)
  ![login-light-active](https://user-images.githubusercontent.com/18715287/61996408-3f96b300-b094-11e9-8071-bdc21f8451f3.png)

- **Login/unlock screens (dark variant):**
  ![login-dark](https://user-images.githubusercontent.com/18715287/61996416-4d4c3880-b094-11e9-859d-77e35695c84f.png)
  ![login-dark-hover](https://user-images.githubusercontent.com/18715287/61996419-53dab000-b094-11e9-92cc-54d47a29b7d5.png)
  ![login-dark-active](https://user-images.githubusercontent.com/18715287/61996421-5a692780-b094-11e9-8c50-36452d2434b8.png)

Some of the text colors in the screens (e.g. user name, etc.) may need to be re-worked slightly, especially for the light variant, but would prefer to see if you agree with changes thus far before proceeding further.

Reply:
After reflecting on this for a little while, the light variant of the unlock/login/lock screens with these changes looks a little jarring and out of place with the overall look of the Arc theme.

It might be a better option to go for a single "dark" look (background) for the gdm screens, with only the panel dropdown menus changing colors between light and dark variants (i.e. similar to what happens with the regular `#panel` styling, which is dark in all variants). This would imply reverting to using `$osd_*` variables for text colors, and re-styling the buttons and entry to use the `osd` versions of respective mixins to achieve a uniform dark look.

Reply:
Thank you for the screenshots. It seems that the login/lock screens haven't been very thoroughly styled in the first place (or the styling may not have been updated and looks off for that reason), so it would be great to fix and polish everything while you're at it.

I do agree that having dark login screen for both light and dark variant is better option. Using the `$osd_*` color variables is the right way to do this, and it looks like lot of the existing code already uses those.

For login/unlock screen buttons you could probably simply use `@extend %osd_button;` for buttons, or if that doesn't work, change the `@include`s to use the osd button versions, as you mentioned.

For the login entry, you could probably do something like this:
```
.login-dialog {
  StEntry {
    @include entry(osd);
    &:focus {
      @include entry(osd-focus);
    }
  }
}
```

Based on the screenshot, for the lock screen panel, I think using `color: $_panel_fg_color;` may be better after all.

IMHO, the transparent panel and notifications on the lock screen don't really make sense for the Arc theme style, but I'm not sure what would be the best solution for those. Only idea that came to mind would be duplicating the regular desktop panel and notification styles for the lock screen as well, but I'm not sure if that would look odd. Happy to hear if you have any ideas for those.